lib/test-rule.js

```diff
@@ -8,8 +8,21 @@ class Reporter {
     this.report = this.report.bind(this);
   }
 
-  report(id, message) {
-    this.messages.push({ id, message });
+  report(id, message, path) {
+
+    let report = {
+      id,
+      message
+    };
+
+    if (path) {
+      report = {
+        path: stringifyPath(path),
```

Comment:
There's no need to stringify the path here. We can stringify the path at any point if we need to.

Reply:
Should we stringify just in the `tableEntry()` then?